chai@2022.03.0 +rocm fails: Imported target "blt_hip" includes non-existent path ... in its INTERFACE_INCLUDE_DIRECTORIES

### Steps to reproduce the issue

Having some trouble building `chai@2022.03.0 +rocm`

Here using Spack-installed ROCm 5.2.3

Concrete spec:

<details><summary>Concretization</summary><pre>
 -   q4wscmv  chai@2022.03.0%gcc@11.1.0~benchmarks~cuda~enable_pick+examples~ipo~openmp~raja+rocm+shared~tests amdgpu_target=gfx90a build_type=RelWithDebInfo arch=linux-ubuntu20.04-x86_64
[+]  y36fzbw      ^blt@0.5.2%gcc@11.1.0 arch=linux-ubuntu20.04-x86_64
[+]  5ihoqvh          ^cmake@3.20.6%gcc@11.1.0~doc+ncurses+ownlibs~qt build_type=Release arch=linux-ubuntu20.04-x86_64
[+]  i2hv5jc              ^ncurses@6.3%gcc@11.1.0~symlinks+termlib abi=none arch=linux-ubuntu20.04-x86_64
[+]  6kp4r3o                  ^pkgconf@1.8.0%gcc@11.1.0 arch=linux-ubuntu20.04-x86_64
[+]  faop6us              ^openssl@1.1.1q%gcc@11.1.0~docs~shared certs=mozilla patches=3fdcf2d arch=linux-ubuntu20.04-x86_64
[+]  t3c4o5i                  ^ca-certificates-mozilla@2022-07-19%gcc@11.1.0 arch=linux-ubuntu20.04-x86_64
[+]  3drusxj                  ^perl@5.34.1%gcc@11.1.0+cpanm+shared+threads arch=linux-ubuntu20.04-x86_64
[+]  fzp4src                      ^berkeley-db@18.1.40%gcc@11.1.0+cxx~docs+stl patches=26090f4,b231fcc arch=linux-ubuntu20.04-x86_64
[+]  3gilvnw                      ^bzip2@1.0.8%gcc@11.1.0~debug~pic+shared arch=linux-ubuntu20.04-x86_64
[+]  6id7p5g                          ^diffutils@3.8%gcc@11.1.0 arch=linux-ubuntu20.04-x86_64
[+]  nxnzjyt                              ^libiconv@1.16%gcc@11.1.0 libs=shared,static arch=linux-ubuntu20.04-x86_64
[+]  u3nqjvs                      ^gdbm@1.19%gcc@11.1.0 arch=linux-ubuntu20.04-x86_64
[+]  tfambxv                          ^readline@8.1.2%gcc@11.1.0 arch=linux-ubuntu20.04-x86_64
[+]  hdtjtng                      ^zlib@1.2.12%gcc@11.1.0+optimize+pic+shared patches=0d38234 arch=linux-ubuntu20.04-x86_64
[+]  zjvnjui      ^hip@5.2.3%gcc@11.1.0~ipo build_type=Release patches=7ed1232 arch=linux-ubuntu20.04-x86_64
[+]  z5vlvra          ^comgr@5.2.3%gcc@11.1.0~ipo build_type=Release arch=linux-ubuntu20.04-x86_64
[+]  psgr6v7              ^llvm-amdgpu@5.2.3%gcc@11.1.0~ipo~link_llvm_dylib~llvm_dylib~openmp+rocm-device-libs build_type=Release patches=a08bbe1 arch=linux-ubuntu20.04-x86_64
[+]  gftl34f                  ^python@3.8.13%gcc@11.1.0+bz2+ctypes+dbm~debug+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tix~tkinter~ucs4+uuid+zlib patches=0d98e93,4c24573,f2fd060 arch=linux-ubuntu20.04-x86_64
[+]  3ns5tna                      ^expat@2.4.8%gcc@11.1.0+libbsd arch=linux-ubuntu20.04-x86_64
[+]  kibpzfz                          ^libbsd@0.11.5%gcc@11.1.0 arch=linux-ubuntu20.04-x86_64
[+]  5zyum6y                              ^libmd@1.0.4%gcc@11.1.0 arch=linux-ubuntu20.04-x86_64
[+]  uswfuz3                      ^gettext@0.21%gcc@11.1.0+bzip2+curses+git~libunistring+libxml2+tar+xz arch=linux-ubuntu20.04-x86_64
[+]  qlk2epp                          ^libxml2@2.10.1%gcc@11.1.0~python arch=linux-ubuntu20.04-x86_64
[+]  gwvbhfp                              ^xz@5.2.5%gcc@11.1.0+pic libs=shared,static arch=linux-ubuntu20.04-x86_64
[+]  2kugi24                          ^tar@1.34%gcc@11.1.0 zip=pigz arch=linux-ubuntu20.04-x86_64
[+]  6jrwfc4                              ^pigz@2.7%gcc@11.1.0 arch=linux-ubuntu20.04-x86_64
[+]  x3a2d4r                              ^zstd@1.5.2%gcc@11.1.0+programs compression=none libs=shared,static arch=linux-ubuntu20.04-x86_64
[+]  5qtg5wc                      ^libffi@3.4.2%gcc@11.1.0 arch=linux-ubuntu20.04-x86_64
[+]  ihcyikh                      ^sqlite@3.39.2%gcc@11.1.0+column_metadata+dynamic_extensions+fts~functions+rtree arch=linux-ubuntu20.04-x86_64
[+]  dsfre7u                      ^util-linux-uuid@2.37.4%gcc@11.1.0 arch=linux-ubuntu20.04-x86_64
[+]  zj7acxm                  ^z3@4.8.16%gcc@11.1.0~gmp~ipo~python build_type=RelWithDebInfo arch=linux-ubuntu20.04-x86_64
[+]  vrt2z4v              ^rocm-cmake@5.2.3%gcc@11.1.0~ipo build_type=Release arch=linux-ubuntu20.04-x86_64
[+]  cpfdgia          ^glx@1.4%gcc@11.1.0 arch=linux-ubuntu20.04-x86_64
[+]  5anzuqn              ^mesa@22.1.2%gcc@11.1.0+glx+llvm+opengl~opengles+osmesa~strip buildtype=release default_library=shared patches=ee737d1 arch=linux-ubuntu20.04-x86_64
[+]  xu7jvvq                  ^bison@3.8.2%gcc@11.1.0 arch=linux-ubuntu20.04-x86_64
[+]  acee3tg                      ^m4@1.4.19%gcc@11.1.0+sigsegv patches=9dc5fbd,bfdffa7 arch=linux-ubuntu20.04-x86_64
[+]  u7hwsy2                          ^libsigsegv@2.13%gcc@11.1.0 arch=linux-ubuntu20.04-x86_64
[+]  ml5tvbh                  ^flex@2.6.3%gcc@11.1.0+lex~nls arch=linux-ubuntu20.04-x86_64
[+]  5mzjae6                      ^findutils@4.9.0%gcc@11.1.0 patches=440b954 arch=linux-ubuntu20.04-x86_64
[+]  iv6zhz3                  ^glproto@1.4.17%gcc@11.1.0 arch=linux-ubuntu20.04-x86_64
[+]  5ihrf5i                      ^util-macros@1.19.3%gcc@11.1.0 arch=linux-ubuntu20.04-x86_64
[+]  ylfxdla                  ^libunwind@1.6.2%gcc@11.1.0~block_signals~conservative_checks~cxx_exceptions~debug~debug_frame+docs+pic+tests+weak_backtrace+xz~zlib components=none libs=shared,static arch=linux-ubuntu20.04-x86_64
[+]  vex4vcl                  ^libx11@1.7.0%gcc@11.1.0 arch=linux-ubuntu20.04-x86_64
[+]  fis6jp6                      ^inputproto@2.3.2%gcc@11.1.0 arch=linux-ubuntu20.04-x86_64
[+]  yhhhztg                      ^kbproto@1.0.7%gcc@11.1.0 arch=linux-ubuntu20.04-x86_64
[+]  pe7dxpq                      ^libxcb@1.14%gcc@11.1.0 arch=linux-ubuntu20.04-x86_64
[+]  ub7ci27                          ^libpthread-stubs@0.4%gcc@11.1.0 arch=linux-ubuntu20.04-x86_64
[+]  t7xel4f                          ^libxau@1.0.8%gcc@11.1.0 arch=linux-ubuntu20.04-x86_64
[+]  w6cnc4k                              ^xproto@7.0.31%gcc@11.1.0 arch=linux-ubuntu20.04-x86_64
[+]  n7q2iq6                          ^libxdmcp@1.1.2%gcc@11.1.0 arch=linux-ubuntu20.04-x86_64
[+]  mref4sa                          ^xcb-proto@1.14.1%gcc@11.1.0 arch=linux-ubuntu20.04-x86_64
[+]  t22vlc2                      ^xextproto@7.3.0%gcc@11.1.0 arch=linux-ubuntu20.04-x86_64
[+]  3mwnaio                      ^xtrans@1.3.5%gcc@11.1.0 arch=linux-ubuntu20.04-x86_64
[+]  24i2cnb                  ^libxext@1.3.3%gcc@11.1.0 arch=linux-ubuntu20.04-x86_64
[+]  4smd4xv                  ^libxt@1.1.5%gcc@11.1.0 arch=linux-ubuntu20.04-x86_64
[+]  mhfncfu                      ^libice@1.0.9%gcc@11.1.0 arch=linux-ubuntu20.04-x86_64
[+]  h52iyaq                      ^libsm@1.2.3%gcc@11.1.0 arch=linux-ubuntu20.04-x86_64
[+]  a3bqx6u                  ^meson@0.63.3%gcc@11.1.0 patches=aa6c50d arch=linux-ubuntu20.04-x86_64
[+]  ytylkug                      ^ninja@1.11.1%gcc@11.1.0 arch=linux-ubuntu20.04-x86_64
[+]  yhbf5kl                      ^py-pip@22.2.2%gcc@11.1.0 arch=linux-ubuntu20.04-x86_64
[+]  5kcuafh                      ^py-setuptools@65.5.0%gcc@11.1.0 arch=linux-ubuntu20.04-x86_64
[+]  s5roi7t                      ^py-wheel@0.37.1%gcc@11.1.0 arch=linux-ubuntu20.04-x86_64
[+]  jvan6xa                  ^py-mako@1.2.2%gcc@11.1.0 arch=linux-ubuntu20.04-x86_64
[+]  svfry3l                      ^py-markupsafe@2.1.1%gcc@11.1.0 arch=linux-ubuntu20.04-x86_64
[+]  ku4fu3r                  ^xrandr@1.5.0%gcc@11.1.0 arch=linux-ubuntu20.04-x86_64
[+]  l63xxu7                      ^libxrandr@1.5.0%gcc@11.1.0 arch=linux-ubuntu20.04-x86_64
[+]  towhy7y                          ^libxrender@0.9.10%gcc@11.1.0 arch=linux-ubuntu20.04-x86_64
[+]  efo3pqc                              ^renderproto@0.11.1%gcc@11.1.0 arch=linux-ubuntu20.04-x86_64
[+]  qqopv4v                          ^randrproto@1.5.0%gcc@11.1.0 arch=linux-ubuntu20.04-x86_64
[+]  xoqomos          ^hsa-rocr-dev@5.2.3%gcc@11.1.0+image~ipo+shared build_type=Release patches=71e6851 arch=linux-ubuntu20.04-x86_64
[+]  yted4k6              ^elfutils@0.186%gcc@11.1.0+bzip2~debuginfod~nls+xz arch=linux-ubuntu20.04-x86_64
[+]  s24cjgv              ^hsakmt-roct@5.2.3%gcc@11.1.0~ipo+shared build_type=Release patches=f926273 arch=linux-ubuntu20.04-x86_64
[+]  2jf2rrl                  ^libdrm@2.4.110%gcc@11.1.0~docs arch=linux-ubuntu20.04-x86_64
[+]  dch343z                      ^libpciaccess@0.16%gcc@11.1.0 arch=linux-ubuntu20.04-x86_64
[+]  tqvtjqs                          ^libtool@2.4.7%gcc@11.1.0 arch=linux-ubuntu20.04-x86_64
[+]  jpprdqh                  ^numactl@2.0.14%gcc@11.1.0 patches=4e1d78c,62fc8a8,ff37630 arch=linux-ubuntu20.04-x86_64
[+]  m4tefbd                      ^autoconf@2.69%gcc@11.1.0 patches=35c4492,7793209,a49dd5b arch=linux-ubuntu20.04-x86_64
[+]  mgplfvc                      ^automake@1.16.5%gcc@11.1.0 arch=linux-ubuntu20.04-x86_64
[+]  ymzwm3g              ^xxd-standalone@8.2.1201%gcc@11.1.0 arch=linux-ubuntu20.04-x86_64
[+]  occp6xt          ^perl-file-which@1.22%gcc@11.1.0 arch=linux-ubuntu20.04-x86_64
[+]  y74btj7          ^perl-uri-encode@1.1.1%gcc@11.1.0 arch=linux-ubuntu20.04-x86_64
[+]  mw2f3sb              ^perl-module-build@0.4224%gcc@11.1.0 arch=linux-ubuntu20.04-x86_64
[+]  qkli3gu          ^rocminfo@5.2.3%gcc@11.1.0~ipo build_type=Release arch=linux-ubuntu20.04-x86_64
[+]  gb5nbls          ^roctracer-dev-api@5.2.3%gcc@11.1.0 arch=linux-ubuntu20.04-x86_64
[+]  xdkb3ud      ^umpire@2022.03.1%gcc@11.1.0+c~cuda+device_alloc~deviceconst+examples~fortran~ipo~numa~openmp+rocm+shared amdgpu_target=gfx90a build_type=RelWithDebInfo patches=78daca9 tests=none arch=linux-ubuntu20.04-x86_64
[+]  czgpgev          ^camp@2022.03.2%gcc@11.1.0~cuda~ipo~openmp+rocm~tests amdgpu_target=gfx90a build_type=RelWithDebInfo arch=linux-ubuntu20.04-x86_64
</pre></details>

Build error:
```
$> spack install -f ./chai-rocm.spec.json
...
==> Installing chai-2022.03.0-q4wscmvwelzooktz2x4tl7s5gnvle54s
==> No binary for chai-2022.03.0-q4wscmvwelzooktz2x4tl7s5gnvle54s found: installing from source
==> Using cached archive: /spack/var/spack/cache/_source-cache/git//LLNL/CHAI.git/v2022.03.0.tar.gz
==> Warning: Fetching from mirror without a checksum!
  This package is normally checked out from a version control system, but it has been archived on a spack mirror.  This means we cannot know a checksum for the tarball in advance. Be sure that your connection to this mirror is secure!
==> No patches needed for chai
==> chai: Executing phase: 'initconfig'
==> chai: Executing phase: 'cmake'
==> Error: ProcessError: Command exited with status 1:
    '/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/cmake-3.20.6-5ihoqvhkfnyjnbbk35nsnnitehbv5oyt/bin/cmake' '-G' 'Unix Makefiles' '-DCMAKE_INSTALL_PREFIX:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/chai-2022.03.0-q4wscmvwelzooktz2x4tl7s5gnvle54s' '-DCMAKE_BUILD_TYPE:STRING=RelWithDebInfo' '-DBUILD_TESTING:BOOL=OFF' '-DCMAKE_INTERPROCEDURAL_OPTIMIZATION:BOOL=OFF' '-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON' '-DCMAKE_INSTALL_RPATH_USE_LINK_PATH:BOOL=ON' '-DCMAKE_INSTALL_RPATH:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/chai-2022.03.0-q4wscmvwelzooktz2x4tl7s5gnvle54s/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/chai-2022.03.0-q4wscmvwelzooktz2x4tl7s5gnvle54s/lib64;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/ncurses-6.3-i2hv5jcfp6vaydwc7s6tbdq4lmy5ab6v/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/openssl-1.1.1q-faop6usmjf4ky4ux7s7atrbvix2c2wo7/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/zlib-1.2.12-hdtjtngxfoicw3xoif5xrdopjdngev3h/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/hip-5.2.3-zjvnjuiw3erfhvd2qzw3rakfiq2mqcmm/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/comgr-5.2.3-z5vlvrasw2sw2klpl5nrdvg4kohwczvh/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/llvm-amdgpu-5.2.3-psgr6v7m2kntdkejtxo4j4jefvonxu74/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/z3-4.8.16-zj7acxmklnnds7joag3behoc4caxcumq/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/mesa-22.1.2-5anzuqnz3wxkrejiyrlpqts3aenv6avy/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/expat-2.4.8-3ns5tnaguutlyxosvb7t54ve5aml633z/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/libbsd-0.11.5-kibpzfzkai6a3xn6tl6qhoco3aghtkyv/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/libmd-1.0.4-5zyum6y7evrhwngejmal3i343loqhubr/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/glproto-1.4.17-iv6zhz3qdep2hn5exwpe55mgl47b3rt7/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/libunwind-1.6.2-ylfxdlar2pkwmfie7bgv2rcq53tvxabt/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/xz-5.2.5-gwvbhfpkxuf4ofixoipzmplz2becfbcx/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/libx11-1.7.0-vex4vclmkjqcmsczcryzq6tlqzsozqvw/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/inputproto-2.3.2-fis6jp6iotupzvz4suln2apqcuacz475/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/kbproto-1.0.7-yhhhztgafwtrqlifrlecqyszxrlldqvo/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/libxcb-1.14-pe7dxpqjmng34e57i2ivcok7pyk2vnmk/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/libpthread-stubs-0.4-ub7ci27op7u74nkypvsawbd22uayffok/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/libxau-1.0.8-t7xel4ftqqiw67oqxvrpapmotrpnkd6q/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/xproto-7.0.31-w6cnc4kdtlfcs2gnipp6f5hhdpfh3uel/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/libxdmcp-1.1.2-n7q2iq6wbdegdvvjyfuh5d5aja6tibxx/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/xcb-proto-1.14.1-mref4safnggln4esixnoat2fbek5xkyw/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/xextproto-7.3.0-t22vlc224fnn6nl4hlufa6dsou6nb7n4/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/libxext-1.3.3-24i2cnbhfa7oyhqex2im67stc7ieakj3/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/libxt-1.1.5-4smd4xveenn4w3fb7ldfnfxu272ulftu/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/libice-1.0.9-mhfncful67kxoovwxzb22hpwodw673t7/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/libsm-1.2.3-h52iyaq2pionogceil4anw23r2jbjcs2/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/util-linux-uuid-2.37.4-dsfre7uumkl6ziuqlpfymfntyvqxomyz/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/libxrandr-1.5.0-l63xxu7fd33qdu6cnfdi3oygdzol5vlz/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/libxrender-0.9.10-towhy7yhuugm36g2k5pyil62czfxhxss/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/renderproto-0.11.1-efo3pqcqcurvwn5hte74kekfevr2u35e/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/randrproto-1.5.0-qqopv4vb5re3gwcxpwu5sgyjuqestfpe/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/hsa-rocr-dev-5.2.3-xoqomosncyzvrmprdijejli63why5lqq/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/elfutils-0.186-yted4k6rhqqczkijyjmt2s7upg5lt3xp/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/bzip2-1.0.8-3gilvnwrzghzcbkmuhcnnmiixzqre6us/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/hsakmt-roct-5.2.3-s24cjgvh64iakepffspeh2qracftzygp/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/libdrm-2.4.110-2jf2rrlx2luabxppcuh5dqqk47dv5zu4/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/libpciaccess-0.16-dch343zp6cgxxm3jjuhb7eh27sejgiqj/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/numactl-2.0.14-jpprdqhedvhvkzkfmbxwecpd6e6dfwoo/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/perl-file-which-1.22-occp6xt67g2ls6hq6u4wsirzs6sfeqeb/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/perl-uri-encode-1.1.1-y74btj7merlpupj6x6xhthkgbekkg2bg/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/umpire-2022.03.1-xdkb3udfksv2w4e7fkdr4i2k4evcbzpe/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/camp-2022.03.2-czgpgevy36if4mthgfnqytmybbvju6mm/lib' '-DCMAKE_PREFIX_PATH:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/umpire-2022.03.1-xdkb3udfksv2w4e7fkdr4i2k4evcbzpe;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/camp-2022.03.2-czgpgevy36if4mthgfnqytmybbvju6mm;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/hip-5.2.3-zjvnjuiw3erfhvd2qzw3rakfiq2mqcmm;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/roctracer-dev-api-5.2.3-gb5nblsum2cefdmybgigc76fjrqvyxvk;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/rocminfo-5.2.3-qkli3gucvr3s2nbiug7ahr7i235ufhq4;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/perl-uri-encode-1.1.1-y74btj7merlpupj6x6xhthkgbekkg2bg;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/perl-file-which-1.22-occp6xt67g2ls6hq6u4wsirzs6sfeqeb;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/hsa-rocr-dev-5.2.3-xoqomosncyzvrmprdijejli63why5lqq;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/hsakmt-roct-5.2.3-s24cjgvh64iakepffspeh2qracftzygp;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/numactl-2.0.14-jpprdqhedvhvkzkfmbxwecpd6e6dfwoo;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/libdrm-2.4.110-2jf2rrlx2luabxppcuh5dqqk47dv5zu4;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/libpciaccess-0.16-dch343zp6cgxxm3jjuhb7eh27sejgiqj;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/elfutils-0.186-yted4k6rhqqczkijyjmt2s7upg5lt3xp;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/glx-1.4-cpfdgiazcq6426r6izpdpekpgqhwwfn4;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/mesa-22.1.2-5anzuqnz3wxkrejiyrlpqts3aenv6avy;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/xrandr-1.5.0-ku4fu3rwbq3ly2zaw6bcvz62g4cqr2rn;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/libxrandr-1.5.0-l63xxu7fd33qdu6cnfdi3oygdzol5vlz;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/randrproto-1.5.0-qqopv4vb5re3gwcxpwu5sgyjuqestfpe;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/libxrender-0.9.10-towhy7yhuugm36g2k5pyil62czfxhxss;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/renderproto-0.11.1-efo3pqcqcurvwn5hte74kekfevr2u35e;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/libxt-1.1.5-4smd4xveenn4w3fb7ldfnfxu272ulftu;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/libsm-1.2.3-h52iyaq2pionogceil4anw23r2jbjcs2;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/libice-1.0.9-mhfncful67kxoovwxzb22hpwodw673t7;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/libxext-1.3.3-24i2cnbhfa7oyhqex2im67stc7ieakj3;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/libx11-1.7.0-vex4vclmkjqcmsczcryzq6tlqzsozqvw;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/xtrans-1.3.5-3mwnaiozpta33d7g4gmpii64cvxkbh3d;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/xextproto-7.3.0-t22vlc224fnn6nl4hlufa6dsou6nb7n4;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/libxcb-1.14-pe7dxpqjmng34e57i2ivcok7pyk2vnmk;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/xcb-proto-1.14.1-mref4safnggln4esixnoat2fbek5xkyw;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/libxdmcp-1.1.2-n7q2iq6wbdegdvvjyfuh5d5aja6tibxx;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/libxau-1.0.8-t7xel4ftqqiw67oqxvrpapmotrpnkd6q;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/xproto-7.0.31-w6cnc4kdtlfcs2gnipp6f5hhdpfh3uel;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/libpthread-stubs-0.4-ub7ci27op7u74nkypvsawbd22uayffok;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/kbproto-1.0.7-yhhhztgafwtrqlifrlecqyszxrlldqvo;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/inputproto-2.3.2-fis6jp6iotupzvz4suln2apqcuacz475;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/libunwind-1.6.2-ylfxdlar2pkwmfie7bgv2rcq53tvxabt;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/glproto-1.4.17-iv6zhz3qdep2hn5exwpe55mgl47b3rt7;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/comgr-5.2.3-z5vlvrasw2sw2klpl5nrdvg4kohwczvh;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/llvm-amdgpu-5.2.3-psgr6v7m2kntdkejtxo4j4jefvonxu74;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/z3-4.8.16-zj7acxmklnnds7joag3behoc4caxcumq;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/util-linux-uuid-2.37.4-dsfre7uumkl6ziuqlpfymfntyvqxomyz;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/xz-5.2.5-gwvbhfpkxuf4ofixoipzmplz2becfbcx;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/expat-2.4.8-3ns5tnaguutlyxosvb7t54ve5aml633z;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/libbsd-0.11.5-kibpzfzkai6a3xn6tl6qhoco3aghtkyv;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/libmd-1.0.4-5zyum6y7evrhwngejmal3i343loqhubr;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/blt-0.5.2-y36fzbweyutouvoq3uzjbloaudnoceb7;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/cmake-3.20.6-5ihoqvhkfnyjnbbk35nsnnitehbv5oyt;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/openssl-1.1.1q-faop6usmjf4ky4ux7s7atrbvix2c2wo7;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/zlib-1.2.12-hdtjtngxfoicw3xoif5xrdopjdngev3h;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/bzip2-1.0.8-3gilvnwrzghzcbkmuhcnnmiixzqre6us;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/ncurses-6.3-i2hv5jcfp6vaydwc7s6tbdq4lmy5ab6v' '-C' '/tmp/root/spack-stage/spack-stage-chai-2022.03.0-q4wscmvwelzooktz2x4tl7s5gnvle54s/spack-src/cabc53f99594-linux-ubuntu20.04-x86_64-gcc@11.1.0.cmake' '/tmp/root/spack-stage/spack-stage-chai-2022.03.0-q4wscmvwelzooktz2x4tl7s5gnvle54s/spack-src'

4 errors found in build log:
     67     -- Enabling all compiler warnings on all targets.
     68     -- Fortran support disabled.
     69     -- CMAKE_C_FLAGS flags are:   -Wall -Wextra
     70     -- CMAKE_CXX_FLAGS flags are:   -Wall -Wextra
     71     -- CMAKE_EXE_LINKER_FLAGS flags are:
     72     -- Configuring done
  >> 73     CMake Error in examples/CMakeLists.txt:
     74       Imported target "blt_hip" includes non-existent path
     75
     76         "/tmp/root/spack-stage/spack-stage-chai-2022.03.0-q4wscmvwelzooktz2x4tl7s5gnvle54s/spack-src/HIP_CLANG_INCLUDE_PATH-NOTFOUND/.."
     77
     78       in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:
     79

     ...

     83
     84       * The installation package was faulty and references files it does not
     85       provide.
     86
     87
     88
  >> 89     CMake Error in examples/CMakeLists.txt:
     90       Imported target "blt_hip" includes non-existent path
     91
     92         "/tmp/root/spack-stage/spack-stage-chai-2022.03.0-q4wscmvwelzooktz2x4tl7s5gnvle54s/spack-src/HIP_CLANG_INCLUDE_PATH-NOTFOUND/.."
     93
     94       in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:
     95

     ...

     99
     100      * The installation package was faulty and references files it does not
     101      provide.
     102
     103
     104
  >> 105    CMake Error in examples/CMakeLists.txt:
     106      Imported target "blt_hip" includes non-existent path
     107
     108        "/tmp/root/spack-stage/spack-stage-chai-2022.03.0-q4wscmvwelzooktz2x4tl7s5gnvle54s/spack-src/HIP_CLANG_INCLUDE_PATH-NOTFOUND/.."
     109
     110      in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:
     111

     ...

     115
     116      * The installation package was faulty and references files it does not
     117      provide.
     118
     119
     120
  >> 121    CMake Error in examples/CMakeLists.txt:
     122      Imported target "blt_hip" includes non-existent path
     123
     124        "/tmp/root/spack-stage/spack-stage-chai-2022.03.0-q4wscmvwelzooktz2x4tl7s5gnvle54s/spack-src/HIP_CLANG_INCLUDE_PATH-NOTFOUND/.."
     125
     126      in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:
```

### Error message

<details><summary>Error message</summary><pre>
... see above
</pre></details>


### Information on your system

* **Spack:** 0.19.0.dev0 (0934c4d602375d9c22794670737478fe66525518)
* **Python:** 3.8.10
* **Platform:** linux-ubuntu20.04-cascadelake
* **Concretizer:** clingo

### Additional information

[spack-build-out.txt](https://github.com/spack/spack/files/9824794/spack-build-out.txt)
[spack-build-env.txt](https://github.com/spack/spack/files/9824795/spack-build-env.txt)
[CMakeOutput.log](https://github.com/spack/spack/files/9824797/CMakeOutput.log)
[CMakeError.log](https://github.com/spack/spack/files/9824798/CMakeError.log)

@davidbeckingsale @wspear 

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

--- a/var/spack/repos/builtin/packages/chai/package.py
+++ b/var/spack/repos/builtin/packages/chai/package.py
@@ -112,6 +112,13 @@ class Chai(CachedCMakePackage, CudaPackage, ROCmPackage):
             self.spec.compiler.version,
         )
 
+    def initconfig_compiler_entries(self):
+        spec = self.spec
+        entries = super(Chai, self).initconfig_compiler_entries()
+        if "+rocm" in spec:
+            entries.insert(0, cmake_cache_path("CMAKE_CXX_COMPILER", spec["hip"].hipcc))
+        return entries
+
     def initconfig_hardware_entries(self):
         spec = self.spec
         entries = super(Chai, self).initconfig_hardware_entries()

